### PR TITLE
enforce the `minimum_zig_version` field in the `build.zig.zon`

### PR DESCRIPTION
**Pull request by Rexicon226**
feel free to change the wording however you'd like :smile:

### CODE REVIEW

File: src/Package/Manifest.zig [hunk 1]
```diff
@@ -248,6 +248,15 @@ const Parse = struct {
                     try appendError(p, main_tokens[field_init], "unable to parse semantic version: {s}", .{@errorName(err)});
                     break :v null;
                 };
+
+                if (p.minimum_zig_version) |min_ver| {
+                    const host_version = @import("builtin").zig_version;
+                    if (host_version.order(min_ver) == .lt) {
+                        try appendError(p, main_tokens[field_init],
+                            \\the zig version you are using is too old to build. host: {}, min: {}.
```
Comment (nektro) [inline src/Package/Manifest.zig:256]: `Your Zig version '{}' does not meet the minimum build requirement of '{}' for package '{s}'`

**Review by nektro — CHANGES_REQUESTED** [review]

File: src/Package/Manifest.zig (hunk 1 above)
Comment (Rexicon226) [reply src/Package/Manifest.zig:256]: Yeah, that's good.

File: lib/init/build.zig.zon [hunk 2]
```diff
@@ -5,8 +5,6 @@
     .version = "0.0.0",
 
     // This field is optional.
-    // This is currently advisory only; Zig does not yet do anything
-    // with this value.
```
Comment (castholm) [inline lib/init/build.zig.zon:9]: For completeness you could also remove the corresponding sentence in https://github.com/ziglang/zig/blob/master/doc/build.zig.zon.md

Comment (Rexicon226) [reply lib/init/build.zig.zon:9]: True! Thanks for noticing.

**Comment by andrewrk** [comment]
I'm not sure an error is the best thing to do here. Let's have a public discussion on the issue tracker about proposed behavior first. This PR is still here in case it should be revived as a result of that discussion.

**Comment by Rexicon226** [comment]
Sounds good!

**Pull request closed without merging**